chapter_11_practic/task4.cpp

```diff
@@ -0,0 +1,109 @@
+// Start with the program of Exercise 1 in this chapter, and add a member function of type
+// bool called isOversize() to the bookand tape classes.Let�s say that a book with more
+// than 800 pages, or a tape with a playing time longer than 90 minutes(which would
+// require two cassettes), is considered oversize.You can access this function from main()
+// and display the string �Oversize� for oversize booksand tapes when you display their
+// other data.If bookand tape objects are to be accessed using pointers to them that are
+// stored in an array of type publication, what do you need to add to the publication
+// base class ? Can you instantiate members of this base class ?
+
+#include <iostream>
+#include <string>
+
+class publication
+{
+protected:
+	std::string name = "Not added!";
+	float price = 0.f;
+public:
+	virtual void getData()
+	{
+		std::cout << "Enter book's name" << std::endl;
+		std::cin >> name;
+		std::cout << "Enter price:" << std::endl;
+		std::cin >> price;
+	}
+	virtual void putData() const
+	{
+		std::cout << "Book's name is: " << name << ", price: " << price;
+	}
+	virtual bool isOversize() = 0;
+};
+
+class book : public publication
+{
+private:
+	int pages = 0;
+public:
+	void getData()
+	{
+		publication::getData();
+		std::cout << "Enter pages count of book" << std::endl;
+		std::cin >> pages;
+	}
+	void putData() const
+	{
+		publication::putData();
+		std::cout << ", pages count of book are: " << pages << std::endl;
+	}
+	bool isOversize()
+	{
+		return (pages > 800 ? true : false);
```

Comment:
you could just write
`return pages > 800`

Reply:
https://github.com/Dmitriy-Romanchuk/master/pull/11/commits/d8ee3c69021f245000775446122ab89b2972cc68